### PR TITLE
COS mapping init

### PR DESCRIPTION
**Pull request by hmcts-shajahan**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)
- [ ] the `keep-helm` label has been added, if the helm release should be persisted after a successful build

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

**Review by channa-patil — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/prl/mapper/citizen/CaseDataMapper.java [hunk 1]
```diff
@@ -128,6 +134,14 @@ public CaseData buildUpdatedCaseData(CaseData caseData) throws JsonProcessingExc
             updateConsentOrderDetailsForCaseData(caseDataBuilder, c100RebuildConsentOrderDetails);
         }
 
+        if (isNotEmpty(c100RebuildData.getC100RebuildSafetyConcerns())) {
+            C100RebuildSafetyConcernsElements c100C100RebuildSafetyConcernsElements = mapper
+                .readValue(c100RebuildData.getC100RebuildSafetyConcerns(), C100RebuildSafetyConcernsElements.class);
+            updateSafetyConcernsElementsForCaseData(caseDataBuilder,
+                                                    c100C100RebuildSafetyConcernsElements,
+                                                    c100RebuildChildDetailsElements.getChildDetails());
```
[the remark this answers is not in the document]
Comment (hmcts-shajahan) [reply src/main/java/uk/gov/hmcts/reform/prl/mapper/citizen/CaseDataMapper.java]: Resolved

File: src/main/java/uk/gov/hmcts/reform/prl/mapper/citizen/CaseDataSafetyConcernsElementsMapper.java [hunk 2]
```diff
@@ -0,0 +1,429 @@
...
+
+        return applicantElements;
+
+    }
+
+    private static ChildAbuse buildChildAbuseDetails(
+        C100RebuildSafetyConcernsElements c100RebuildSafetyConcernsElements,ChildAbuseEnum abuseType) {
+        ChildSafetyConcernsDto childAbuse = c100RebuildSafetyConcernsElements.getC100SafetyConcerns().getChild();
+
+        if (isNotEmpty(childAbuse.getPhysicalAbuse()) && abuseType.equals(ChildAbuseEnum.physicalAbuse)) {
```
[the remark this answers is not in the document]
Comment (hmcts-shajahan) [reply src/main/java/uk/gov/hmcts/reform/prl/mapper/citizen/CaseDataSafetyConcernsElementsMapper.java]: resolved

File: src/main/java/uk/gov/hmcts/reform/prl/mapper/citizen/CaseDataSafetyConcernsElementsMapper.java [hunk 3]
```diff
@@ -0,0 +1,429 @@
...
+
+    public static void updateSafetyConcernsElementsForCaseData(CaseData.CaseDataBuilder caseDataBuilder,
+                                                               C100RebuildSafetyConcernsElements c100RebuildSafetyConcernsElements,
+                                                               List<ChildDetail> childDetails) {
+        caseDataBuilder.allegationOfHarmRevised(buildAllegationOfHarmRevised(c100RebuildSafetyConcernsElements,childDetails));
+        System.out.println(buildAllegationOfHarmRevised(c100RebuildSafetyConcernsElements,childDetails));
+    }
+
+    private static AllegationOfHarmRevised buildAllegationOfHarmRevised(C100RebuildSafetyConcernsElements c100RebuildSafetyConcernsElements,
+                                                                        List<ChildDetail> childDetails) {
```
[the remark this answers is not in the document]
Comment (hmcts-shajahan) [reply src/main/java/uk/gov/hmcts/reform/prl/mapper/citizen/CaseDataSafetyConcernsElementsMapper.java]: resolved

File: src/main/java/uk/gov/hmcts/reform/prl/mapper/citizen/CaseDataSafetyConcernsElementsMapper.java [hunk 4]
```diff
@@ -0,0 +1,429 @@
...
+            applicantElements.add(mapToDomesticAbuse(TypeOfAbuseEnum.TypeOfAbuseEnum_value_5, applicantAbuse.getFinancialAbuse()));
+        }
+        if (isNotEmpty(applicantAbuse.getSexualAbuse())) {
+            applicantElements.add(mapToDomesticAbuse(TypeOfAbuseEnum.TypeOfAbuseEnum_value_3, applicantAbuse.getSexualAbuse()));
+        }
+        if (isNotEmpty(applicantAbuse.getPsychologicalAbuse())) {
+            applicantElements.add(mapToDomesticAbuse(TypeOfAbuseEnum.TypeOfAbuseEnum_value_2, applicantAbuse.getPsychologicalAbuse()));
+        }
+
+        return applicantElements;
```
[the remark this answers is not in the document]
Comment (hmcts-shajahan) [reply src/main/java/uk/gov/hmcts/reform/prl/mapper/citizen/CaseDataSafetyConcernsElementsMapper.java:362]: resolved

**Review by channa-patil — APPROVED** [review]

**Pull request merged**